src/Language/Fixpoint/Solver/PLE.hs

```diff
@@ -460,6 +464,12 @@ data EvalEnv = EvalEnv
   , evSMTCache :: M.HashMap Expr Bool -- ^ Whether an expression is valid or its negation
   , evFuel     :: FuelCount
   , extensionalityFlag :: Bool -- ^ True if the extensionality flag is turned on
+  
+  -- Eta expansion feature
+  , freshEtaNames :: Int -- ^ Keeps track of how many names we generated to perform eta 
+                         --   expansion, we use this to generate always fresh names
+  , hoFlag        :: Bool -- ^ True if the allowho flag is turned on, needed for the eta
+                          -- expansion reasoning as its going to generate ho constraints
```

Comment:
This comment would need to point to a more precise explanation of what such reasoning is. It could be done with a longer note at the top-level.

From what I can see, it is just enabling beta reduction of lambdas and eta expansion of functions which appear in non-saturated applications.